moabb/paradigms/motor_imagery.py

```diff
@@ -69,21 +69,22 @@ def process_raw(self, raw, dataset):
 
         # get event id
         if self.events is None:
-            event_id = dataset.event_id
+            event_id = dataset.selected_events
```

Comment:
See my comment. If we restrict to a certain set of events, this done by the paradigm. the dataset is context independent.